### PR TITLE
Pull request for week0002

### PR DESCRIPTION
**Pull request by dbugxpert**

### CODE REVIEW

File: week0001/dbugxpert/week0001.py [hunk 1]
```diff
@@ -0,0 +1,5 @@
+for x in range(1, 101):
```
Comment (vietlq) [inline week0001/dbugxpert/week0001.py]: This looks good.

File: week0002/dbugxpert/python/week0002.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
+n = 4
```
Comment (vietlq) [inline week0002/dbugxpert/python/week0002.py]: Use single line assignment: `n, m = 4, 4`

File: week0002/dbugxpert/python/week0002.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+m = 4
+g = [[0 for x in range(n)] for y in range(m)]
```
Comment (vietlq) [inline week0002/dbugxpert/python/week0002.py]: What are `g tr br lc rc x`? Give them meaningful names!

File: week0002/dbugxpert/python/week0002.py (hunk 2 above)
Comment (vietlq) [inline week0002/dbugxpert/python/week0002.py]: Also give a comment what are `n` & `m`. No one understands what are they.

File: week0002/dbugxpert/python/week0002.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
+n = 4
+m = 4
+g = [[0 for x in range(n)] for y in range(m)] 
+tr = 0
+br = m-1
+lc = 0
+rc = n-1
```
Comment (vietlq) [inline week0002/dbugxpert/python/week0002.py]: Space around operators`+ - * / // += -=`

File: week0002/dbugxpert/python/week0002.py [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+  rc -= 1
+  for t in range(rc, lc-1, -1):  
+    g[br][t] = x
+    x += 1
+  br -= 1
+  for t in range(br, tr-1, -1):
+    g[t][lc] = x
+    x += 1 
+  lc += 1  
+l = len(str(n*m))
```
Comment (vietlq) [inline week0002/dbugxpert/python/week0002.py]: Space around operators `+ - * / // += -=`

File: week0002/dbugxpert/python/week0002.py [hunk 6]
```diff
@@ -1,28 +1,29 @@
...
-rc = n-1
-x = 1
-for _ in range(m//2):
-  for t in range(lc, rc+1):
-    g[tr][t] = x
+n, m = 4, 4 # input for size of grid
+# comment out the following line for input parameters if required
+# n, m = map(int, input().strip().split()) 
+grid = [[0 for x in range(n)] for y in range(m)]
+top_row = 0
```
Comment (vietlq) [inline week0002/dbugxpert/python/week0002.py]: You can put `top_row` and friends on 1 line as well

File: week0002/dbugxpert/python/week0002.py [hunk 7]
```diff
@@ -1,28 +1,29 @@
...
+  bottom_row -= 1
+  for t in range(bottom_row, top_row - 1, -1):
+    grid[t][left_column] = x
     x += 1 
-  lc += 1  
-l = len(str(n*m))
-for x in g:
+  left_column += 1    
+l = len(str(n*m)) # max len will be used for padding
+for x in grid:
```
Comment (vietlq) [inline week0002/dbugxpert/python/week0002.py]: For row in grid.

File: week0002/dbugxpert/python/week0002.py [hunk 8]
```diff
@@ -1,28 +1,29 @@
...
+n, m = 4, 4 # input for size of grid
+# comment out the following line for input parameters if required
+# n, m = map(int, input().strip().split()) 
+grid = [[0 for x in range(n)] for y in range(m)]
+top_row = 0 
+bottom_row = m-1
+left_column = 0
+right_column = n-1
+x = 1 # counter for value that will be printed
+for _ in range(m // 2): # loop around the grid
```
Comment (vietlq) [inline week0002/dbugxpert/python/week0002.py]: Comment above the statement as well

File: week0002/dbugxpert/python/week0002.py [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+  right_column -= 1
+  for t in range(right_column, left_column - 1, -1):  
+    grid[bottom_row][t] = x
+    x += 1
+  bottom_row -= 1
+  for t in range(bottom_row, top_row - 1, -1):
+    grid[t][left_column] = x
+    x += 1 
+  left_column += 1    
+l = len(str(n*m)) # max len will be used for padding
```
Comment (vietlq) [inline week0002/dbugxpert/python/week0002.py]: Give `l` a more explicit name, say MAX_NUM_LEN

File: week0002/dbugxpert/python/week0002.py [hunk 10]
```diff
@@ -0,0 +1,30 @@
+# input size for grid
+n, m = 4, 4 
+# n, m = map(int, input().strip().split())
```
Comment (vietlq) [inline week0002/dbugxpert/python/week0002.py]: This is cheeky but looks fun hehe

File: week0002/dbugxpert/python/week0002.py [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+  right_column -= 1
+  for t in range(right_column, left_column - 1, -1):  
+    grid[bottom_row][t] = counter
+    counter += 1
+  bottom_row -= 1
+  for t in range(bottom_row, top_row - 1, -1):
+    grid[t][left_column] = counter
+    counter += 1 
+  left_column += 1
+padding_size = len(str(n*m))
```
Comment (vietlq) [inline week0002/dbugxpert/python/week0002.py]: I recommend UPPER_CASE for constants like `padding_size`.

File: week0002/dbugxpert/python/week0002.py [hunk 12]
```diff
@@ -0,0 +1,30 @@
...
+  for t in range(right_column, left_column - 1, -1):  
+    grid[bottom_row][t] = counter
+    counter += 1
+  bottom_row -= 1
+  for t in range(bottom_row, top_row - 1, -1):
+    grid[t][left_column] = counter
+    counter += 1 
+  left_column += 1
+padding_size = len(str(n*m)) 
+for x in grid:
```
Comment (vietlq) [inline week0002/dbugxpert/python/week0002.py]: x => row
y => cell

**Comment by vietlq** [comment]
Don't worry, you are doing good.

File: week0002/dbugxpert/python/week0002.py (hunk 11 above)
Comment (vietlq) [inline week0002/dbugxpert/python/week0002.py]: Also put a blank line after the `for` loop or readability

**Comment by dbugxpert** [comment]
Thanks for all the comments, really appreciated.

**Comment by vietlq** [comment]
Awesome! Not a single error!

**Pull request merged**